src/main/scala/mesosphere/marathon/core/health/impl/MarathonHealthCheckManager.scala

```diff
@@ -51,6 +55,32 @@ class MarathonHealthCheckManager(
       ahcs(appId).values.flatten.toSet
     }
 
+  /**
+    * A common materialized merge hub that is used by all [[HealthCheckActor]]s to channel the Marathon health checks.
+    * Its main purpose is to provide a global throttling mechanism that limit a total number of concurrent Marathon
+    * health check in the system defined by [[MarathonConf.maxConcurrentMarathonHealthChecks]] parameter. After
+    * executing the health check by calling the [[HealthCheckWorker.run()]] method the result is sent back to the
+    * HealthCheckActor via the provided ActorRef.
+    */
+  val healthCheckWorkerHub: Sink[(AppDefinition, Instance, MarathonHealthCheck, ActorRef), NotUsed] =
+    MergeHub
+      .source[(AppDefinition, Instance, MarathonHealthCheck, ActorRef)](1)
+      .mapAsync(conf.maxConcurrentMarathonHealthChecks()){
+        case (app, instance, marathonHealthCheck, actorRef) =>
+          HealthCheckWorker
+            .run(app, instance, marathonHealthCheck)
+            .map(healthResult => actorRef ! healthResult)
```

Comment:
Hm, I'm always confused by `map` calls with side effects. What do you think about using a `Sink.foreach` below instead? Also, we won't have backpressure here if the actors cannot keep up, right?

Reply:
This is `Future.map`, not the `Stream.map`. Also, keep in mind that we're inside the `mapAsync` stage so we have to return a `Future[_]`. As for the backpressure - the `healthResult` is sent back to the `HealthCheckActor` that produced the health check so we're fine here which will not produce another one if the merge hub is full.

Reply:
The stream in `HealthCheckActor` keeps producing health checks even if the actor itself cannot process the results of health checks. This could wait for an ack from the actor. This would yield backpressure if the actor cannot process health check results and would somewhat capture the sideeffect. Still ugly, though.

Reply:
I see what you saying. Yes, this is unlikely but theoretically possible. Anyway, this is the case for any scheduled actor-actor message interaction. This PR solves a different issue.